app/domain/authentication/authn_k8s/k8s_resolver.rb

```diff
@@ -74,91 +65,127 @@ def validate_pod
       # Tests whether the Pod's ReplicaSet belongs to the Deployment.
       class Deployment < Base
         def validate_pod
-          replica_set_ref = verify "Pod #{pod_name} does not belong to a ReplicaSet (or Deployment)" do
-            pod_owner_refs &&
-              pod_owner_refs.find{|ref| ref.kind == "ReplicaSet"}
+          replica_set_ref = pod_owner_refs && pod_owner_refs.find { |ref| ref.kind == "ReplicaSet" }
```

Comment:
It's a little odd that we're checking whether `pod_owner_refs` is `nil` as part of the assignment operation here. We can instead use the safe navigation operator and make this:

```
replica_set_ref = pod_owner_refs&.find { |ref| ref.kind == "ReplicaSet" }
```

 Same with all the other similar cases below.

Reply:
will do!